### ISSUE TITLE
Unable to modify xml reports

### ISSUE THREAD

**Issue by FuriousDuck** [pytest-dev/pytest#2381]
It seems that [this commit](https://github.com/pytest-dev/pytest/commit/d3a6be413014b112227cbdb489c62483ec4d9192) has broken an ability to modify stderr(out) via reporting hooks in xml reports which was pretty useful (because xml reports do not attach custom sections and record_xml_property doesn't work with pytest-xdist)

This code
```
@pytest.hookimpl(hookwrapper=True)
def pytest_runtest_makereport(item, call):
    outcome = yield
    report = outcome.get_result()
    if report.when == 'call':
        if report.skipped and 'xfail' in report.keywords or report.failed and 'xfail' not in report.keywords:
            try:
                fail_info = 'Fail detailed info:\n' + get_fail_info(item)
            except WebDriverException:
                fail_info = 'Fail detailed info:\n\nCaught a webdriver exception. Test session must have failed'
            report.sections.append(('Captured stdout call', fail_info))
```
is adding some strings to 'Captured stdout call' section of report. After this commit, strings are still visible in console log, but not in xml

**Comment by RonnyPfannschmidt** [comment]
@KKoukiou can you take a look at this?

**Comment by KKoukiou** [comment]
@FuriousDuck @RonnyPfannschmidt I took a look at this issue. 
Please replace `report.sections.append(('Captured stdout call', fail_info))` in your code
with: `item.add_report_section(report.when, "stdout", fail_info)` and tell me how it works.
It worked for me fine.

Thanks,
Katerina

**Comment by nicoddemus** [comment]
Thanks for looking into it @KKoukiou.

**Comment by nicoddemus** [comment]
@FuriousDuck any feedback?

**Comment by timyhou** [comment]
Glad I came across this. I was following this [StackOverflow post](https://stackoverflow.com/questions/29526054/customizing-pytest-junitxml-failure-reports) as a guide. I also experienced the same issue with trying to append to `Captured stdout` and `Captured stderr` like the OP described. 

However, @KKoukiou suggestion worked for me for stdout or stderr. This is a gem. I couldn't find any documentation describing the right way to modify report sections.
